### DISCUSSION TITLE
Generate custom SDKs for langfuse prompts

### DISCUSSION THREAD

**Discussion by zachkirsch** [langfuse/langfuse#4766] [Ideas]
### Describe the feature or potential improvement

It would be awesome if langfuse generated a custom SDK for my langfuse prompts. Between prompt names and prompt params, there's a lot of magic strings floating around my codebase right now, and zero type safety. This would improve the devx and reduce bugs at runtime.

**Comment by zachkirsch** [comment]
Publishing SDKs feels ideal, but a quicker & dirtier solution would to just generate a whole `langfuse.ts` for my prompts that I can copy from the Langfuse UI and paste into my codebase, e.g.

```ts
// generated code
// copied from langfuse ui

export const LangfusePrompts {
  myPrompt1: {
    compile: async (arg1: string) => { /* ... */ },
  },
  myPrompt2: {
    compile: async (arg1: string, arg2: string) => { /* ... */ },
  },
  /* ... */
}
```

**Reply by marcklingen** [reply]
That'd be cool!

I considered making get_prompt generic and offering copy-paste snippets on each prompt view in the UI. Being able to download a single langfuse.ts file is better for TS for sure. Are you interested in contributing this after discussing more details on an issue?

**Reply by zachkirsch** [reply]
I honestly prob don't have the bandwidth to contribute rn :/ but if/when I do I would love to contribute

**Reply by marcklingen** [reply]
sounds good!

**Comment by marcklingen** [comment]
+1, agree that the magic strings are not the best DX right now, both for prompt names, the types that these prompts have (chat or text) and the variables that are available for each prompt.

Options
- SDK authenticates with API and can generate type file locally (complicated)
- Export type files (as [suggested](https://github.com/orgs/langfuse/discussions/4766#discussioncomment-11601190) by @zachkirsch)
- get_prompt generic that accepts the type. Add snippet to langfuse UI to make it easy to copy/paste this for every prompt.
